nats.go

```diff
@@ -2026,33 +2018,32 @@ func (nc *Conn) Request(subj string, data []byte, timeout time.Duration) (*Msg,
 	}
 
 	// snapshot
-	var doSetup, useOldRequestStyle bool
 	nc.mu.Lock()
-	useOldRequestStyle = nc.Opts.UseOldRequestStyle
-	doSetup = (nc.respMux == nil)
+	useOldRequestStyle := nc.Opts.UseOldRequestStyle
 	nc.mu.Unlock()
 
 	// If user wants the old style.
 	if useOldRequestStyle {
 		return nc.oldRequest(subj, data, timeout)
 	}
 
-	// Make sure scoped subscription is setup at least once on first
-	// call to Request(). Will handle duplicates in createRespMux.
-	if doSetup {
-		if err := nc.createRespMux(); err != nil {
-			return nil, err
-		}
+	// Make sure scoped subscription is setup only once on first call to
+	// Request().
+	var err error
+	nc.respSetup.Do(func() { err = nc.createRespMux() })
```

Comment:
The Once struct and hence the Do has both a mutex and atomic for fast path evaluation for when the function has already been called. But this again is mixing and matching the Once mutex with the Connection mutex. But that may be ok in this case.